Mathlib/Algebra/FreeMonoid/Basic.lean

```diff
@@ -415,4 +423,41 @@ theorem length_reverse {a : FreeMonoid α} : a.reverse.length = a.length :=
 
 end Reverse
 
+section IsomorphicTypes
+
+variable {α β : Type*}
+
+/-- free monoids over isomorphic types are isomorphic -/
+@[to_additive "if two types are isomorphic, the additive free monoids over those types are
+isomorphic"]
+def congrEquiv (e : α ≃ β) : FreeMonoid α ≃* FreeMonoid β :=
+  MulEquiv.mk' ⟨FreeMonoid.map e.toFun, FreeMonoid.map e.invFun, fun _ => map_invFun_map_toFun_eq e,
+    fun _ => map_toFun_map_invFun_eq e⟩ (by simp [map_mul])
+
+@[to_additive (attr := simp)]
+theorem congrEquiv_one (e : α ≃ β) : congrEquiv e (1) = 1 := rfl
```

Comment:
Because `congrEquiv` is a `MulEquiv`, this lemma is in fact a bit superfluous. `(congrEquiv e).map_one` already gives it.

Sorry for the yoyo effect, but could you please drop this one, and the `_mul` lemma?

Reply:
sure thing!